finalfusion-ffi/src/lib.rs

```diff
@@ -65,3 +71,42 @@ pub unsafe extern "C" fn ff_free_embeddings(embeddings: *mut Embeddings<VocabWra
         Box::from_raw(embeddings);
     }
 }
+
+/// Return the embedding dimensionality.
+#[no_mangle]
+pub unsafe extern "C" fn dims(embeddings: *const Embeddings<VocabWrap, StorageWrap>) -> usize {
+    check_null!(embeddings);
+    let embeddings = &*embeddings;
+    embeddings.storage().shape().1
+}
+
+/// Embedding Lookup.
+///
+/// Writes the embedding to `buffer`. Callers need to verify that `buffer` is large enough for the
+/// embedding dimensionality.
+///
+/// Returns an integer indicating whether an embedding could be found and anything was written to
+/// `buffer`.
+///
+/// **Note:** Passing a buffer with incorrect size will lead to UB. The correct dimensionality can
+/// be obtained through `dims()`.
+#[no_mangle]
+pub unsafe extern "C" fn embedding(
+    embeddings: *const Embeddings<VocabWrap, StorageWrap>,
+    word: *const c_char,
+    buffer: *mut f32,
+) -> usize {
+    check_null!(embeddings);
+    check_null!(word);
+    check_null!(buffer);
+
+    let embeddings = &*embeddings;
+    let word = CStr::from_ptr(word).to_string_lossy();
+    if let Some(embedding) = embeddings.embedding(word.as_ref()) {
```

Comment:
Is `as_ref` necessary here or does `&word` work?

Reply:
`&word` works, not sure why I used `as_ref`